connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -332,6 +333,12 @@ public HeaderConverter newHeaderConverter(AbstractConfig config, String classPro
         if (plugin == null) {
             throw new ConnectException("Unable to instantiate '" + klassName + "'");
         }
+        if (plugin instanceof Versioned) {
+            Versioned versionedPlugin = (Versioned) plugin;
+            if (versionedPlugin.version() == null || versionedPlugin.version().trim().isEmpty()) {
+                throw new ConnectException("Version not defined for '" + klassName + "'");
+            }
```

Comment:
For 2.0, should we log this as an error rather than throw an exception? IIUC, this is what you suggested, @ewencp.

Reply:
My understanding was that we enforce it strictly for RestExtension in 2.0. Older implementations can be enforced in 3.0

Reply:
I think there's confusion in the discussion. I think @rhauch's point is that, since we have not enforced correctness in returned values thus far, we should be liberal in what we accept for now. So if they return `null` or an empty value, we should log an error, but not throw an exception that would kill the connector.

On 3.0 or some later version, we'd do as this code currently does and throw an exception since we will have given connectors a reasonable grace period to fix their behavior given that we didn't previously enforce the behavior.